python-threatexchange/threatexchange/dataset.py

```diff
@@ -106,3 +110,23 @@ def load_cache(
                 signal_type.load(signal_state_file)
             ret.append(signal_type)
         return ret
+
+    def store_index(self, signal_type: signal_base.SignalType, index) -> None:
+        if not self.state_dir.exists():
+            self.state_dir.mkdir()
+        path = self._index_file(signal_type)
+        if index is None:
+            if path.exists():
+                path.unlink()
+            return
+        with path.open("wb") as fout:
```

Comment:
Caveat: This makes row based formats like CSV harder to use on Windows. 

Not that anyone is trying to do it, but flagging.